packages/connections/src/ConnectionEvent.js

```diff
@@ -0,0 +1,46 @@
+const OPEN = Symbol("OPEN");
+const DATA = Symbol("DATA");
+const ERROR = Symbol("ERROR");
+const COMPLETE = Symbol("COMPLETE");
+const ABORT = Symbol("ABORT");
+
+export default class ConnectionEvent {
+  /**
+   * Connection event
+   * @param {EventEmitter} target
+   * @param {Symbol} type
+   */
+  constructor(target, type) {
+    /**
+     * @property {EventEmitter}
+     * @name ConnectionEvent#target
+     */
+    Object.defineProperty(this, "target", { value: target });
+
+    /**
+     * @property {Symbol}
+     * @name ConnectionEvent#type
+     */
+    Object.defineProperty(this, "type", { value: type });
+  }
+
+  static get OPEN() {
+    return OPEN;
+  }
+
+  static get DATA() {
+    return DATA;
+  }
+
+  static get ERROR() {
+    return ERROR;
+  }
+
+  static get COMPLETE() {
+    return COMPLETE;
```

Comment:
I know thinking of an Request `COMPLETE` feels more natural, but as we are talking about Connections here, shouldnt this be `CLOSED`? Also it would be in line with `AbstractConnection.CLOSED`

Reply:
There is a difference between the states (INIT, OPEN, CLOSED) and the events which convey reason.

Reply:
done 👍